Sources/MongoSwift/BSON/Document.swift

```diff
@@ -24,10 +24,14 @@ public class DocumentStorage {
         self.pointer = bson_new()
     }
 
-    internal init(fromPointer pointer: BSONPointer) {
+    internal init(copying pointer: BSONPointer) {
         self.pointer = bson_copy(pointer)
     }
 
+    internal init(stealing pointer: MutableBSONPointer) {
+        self.pointer = pointer
```

Comment:
what if `self.pointer` is already set?

Reply:
since this is an initializer, I don't think that is possible.

Reply:
that's a... really good point 😄 I'm like a hammer looking for nails over here